src/linalg.jl

```diff
@@ -379,36 +315,37 @@ _norm_p0(x) = x == 0 ? zero(x) : one(x)
     end
 end
 
-@inline Base.normalize(a::StaticVector) = inv(vecnorm(a))*a
-@inline Base.normalize(a::StaticVector, p::Real) = inv(vecnorm(a, p))*a
+@inline normalize(a::StaticVector) = inv(vecnorm(a))*a
+@inline normalize(a::StaticVector, p::Real) = inv(vecnorm(a, p))*a
 
-@inline Base.normalize!(a::StaticVector) = (a .*= inv(vecnorm(a)))
-@inline Base.normalize!(a::StaticVector, p::Real) = (a.*= inv(vecnorm(a, p)))
+@inline normalize!(a::StaticVector) = (a .*= inv(vecnorm(a)))
+@inline normalize!(a::StaticVector, p::Real) = (a .*= inv(vecnorm(a, p)))
 
-
-@generated function trace(a::StaticMatrix)
-    s = size(a)
-    if s[1] != s[2]
-        error("matrix is not square")
+@inline trace(a::StaticMatrix) = _trace(Size(a), a)
+@generated function _trace(::Size{S}, a::StaticMatrix) where {S}
+    if S[1] != S[2]
+        throw(DimensionMismatch("matrix is not square"))
     end
 
-    if s[1] == 0
+    if S[1] == 0
         return zero(eltype(a))
     end
 
-    exprs = [:(a[$(sub2ind(s, i, i))]) for i = 1:s[1]]
+    exprs = [:(a[$(sub2ind(S, i, i))]) for i = 1:S[1]]
     total = reduce((ex1, ex2) -> :($ex1 + $ex2), exprs)
```

Comment:
Are we sure that multi-argument + is performant? (There will be a splat somewhere, which potentially ruins performance for vectors and matrices with 16+ elements)